lib/resty/upstream/healthcheck.lua

```diff
@@ -435,6 +474,49 @@ local function get_lock(ctx)
     return true
 end
 
+local function set_all_down_flag(ctx, ppeers, bpeers)
+    local dict = ctx.dict
+    local u = ctx.upstream
+    local u_key = gen_upstream_key(u,"all_down")
+
+    local all_down, err = dict:get(u_key)
+    if not all_down then
+        errlog("Failed to get all down flag for upstream " .. u .. ". ", err)
+        return
+    end
+    
+    if all_down == 0 then
+        local result = true
+        
+        for i = 1, #ppeers do 
+            result = result and ppeers[i].down
+        end
+
+        for j = 1, #bpeers do
+            result = result and bpeers[j].down
+        end
+
+        if result then
+            local ok, err = dict:set(u_key, 1)
+            if not ok then
+                errlog("Failed to set all down flag for upstream " .. u .. ". ", err)
+            end
+            
+            for i = 1, #ppeers do
+                set_peer_down_globally(ctx, false, ppeers[i].id, nil)
+                --Flush local cache
```

Comment:
Please always use a space after `--` and avoid using upper case letter at the beginning of a comment text.

Reply:
Fixed